hash-tables/problem-set-hash-tables.md

```diff
@@ -0,0 +1,287 @@
+        self.assertEqual(get_intersection(red, blue), [])
+
+    def test_get_intersection_many_matches(self):
+        red = [50, 43, 25, 72]
+        blue = [25, 36, 43, 50, 80]
+        result = get_intersection(red, blue)
+        self.assertTrue(25 in result)
+        self.assertTrue(43 in result)
+        self.assertTrue(50 in result)
+        self.assertEqual(len(result), 3)
+```
+### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+def get_intersection(red, blue):
+    frequency_hash = {}
+
+    for list in [red, blue]:
+        for item in list:
+            if frequency_hash.get(item):
+                frequency_hash[item] += 1
+            else:
+                frequency_hash[item] = 1
+
+    intersections = []
+    for item, quantity in frequency_hash.items():
+        if quantity > 1:
+            intersections.append(item)
+
+    return intersections
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: LbtYXn
+* title: Hash Tables
+* points: 3
+### !question
+
+Write a function `is_permutation` that takes two strings and returns `True` if one string is a permutation of the other.
+
+Here are the tests:
+
+```python
+def test_is_permutation():
+    assert is_permutation("hello", "ehllo")
+
+
+def test_is_permutation_false_different_qty_letters():
+    assert not is_permutation("heelo", "ehllo")
+
+
+def test_is_permutation_false_different_letters():
+    assert not is_permutation("pizza", "pasta")
+
+
+def test_is_permutation_unmatching_lengths():
+    assert not is_permutation("pizza", "piza")
+
+
+def test_is_permutation_same_word():
+    assert is_permutation("pizza", "pizza")
+
+
+def test_is_permutation_empty_strings():
+    assert is_permutation("", "")
+```
+
+### !end-question
+### !tests
+```python
+import unittest
+from main import *
+class TestChallenge(unittest.TestCase):
+    # Note: Written with these asserts to mimic the pytest assertions we give students
+
+    def test_is_permutation(self):
+        self.assertTrue(is_permutation("hello", "ehllo"))
+
+    def test_is_permutation_false_different_qty_letters(self):
+        self.assertTrue(not is_permutation("heelo", "ehllo"))
+
+    def test_is_permutation_false_different_letters(self):
+        self.assertTrue(not is_permutation("pizza", "pasta"))
+
+    def test_is_permutation_unmatching_lengths(self):
+        self.assertTrue(not is_permutation("pizza", "piza"))
+
+    def test_is_permutation_same_word(self):
+        self.assertTrue(is_permutation("pizza", "pizza"))
+
+    def test_is_permutation_empty_strings(self):
+        self.assertTrue(is_permutation("", ""))
+```
+### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+def populate_frequency_map(text):
+    frequency_hash = {}
+    for char in text:
+        if frequency_hash.get(char):
```

Comment:
similar minimal change here, but consider using `.get(char, 0)` as in comment above
```suggestion
        if char in frequency_hash:
```